app/src/test/java/com/mapbox/navigation/examples/basics/WaypointsSetTest.kt

```diff
@@ -0,0 +1,74 @@
+package com.mapbox.navigation.examples.basics
+
+import com.mapbox.geojson.Point
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class WaypointsSetTest {
+    @Test
+    fun `map not silent waypoints types to waypoint indices`() {
+        val set = WaypointsSet().apply {
+            addRegular(createPoint())
+            addRegular(createPoint())
+            addNamed(createPoint(), "test")
+        }
+
+        val indexes = set.waypointsIndices()
+
+        assertEquals(listOf(0, 1, 2), indexes)
+    }
+
+    @Test
+    fun `map silent waypoints to indices`() {
+        val set = WaypointsSet().apply {
+            addRegular(createPoint())
+            addSilent(createPoint())
+            addNamed(createPoint(), "test")
+        }
+
+        val indexes = set.waypointsIndices()
+
+        assertEquals(listOf(0, 2), indexes)
+    }
+
+    @Test
+    fun `map waypoints to names`() {
+        val set = WaypointsSet().apply {
+            addRegular(createPoint())
+            addRegular(createPoint())
+            addNamed(createPoint(), "test named")
+        }
+
+        val names = set.waypointsNames()
+
+        assertEquals(listOf("", "", "test named"), names)
+    }
+
+    @Test
+    fun `map regular, silent, and named waypoints to names`() {
+        val set = WaypointsSet().apply {
+            addRegular(createPoint())
+            addSilent(createPoint())
+            addNamed(createPoint(), "test named")
+        }
+
+        val names = set.waypointsNames()
+
+        assertEquals(listOf("", "test named"), names)
+    }
```

Comment:
That's actually a tricky case. Silent points should be ignored in the names section. Did I understand it correct?

Reply:
yes, that's right. Silent waypoints cannot be named